service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/validator/ManageOrderEndDateWithEndOfProceedingsValidatorTest.java

```diff
@@ -0,0 +1,247 @@
+package uk.gov.hmcts.reform.fpl.service.orders.validator;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.event.ManageOrdersEventData;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.orders.ManageOrdersEndDateTypeWithEndOfProceedings.CALENDAR_DAY;
+import static uk.gov.hmcts.reform.fpl.enums.orders.ManageOrdersEndDateTypeWithEndOfProceedings.CALENDAR_DAY_AND_TIME;
+import static uk.gov.hmcts.reform.fpl.enums.orders.ManageOrdersEndDateTypeWithEndOfProceedings.END_OF_PROCEEDINGS;
+import static uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock.MANAGE_ORDER_END_DATE_WITH_END_OF_PROCEEDINGS;
+
+class ManageOrderEndDateWithEndOfProceedingsValidatorTest {
+    private static final String TEST_AFTER_APPROVAL_DATE_MESSAGE = "Enter an end date after the approval date";
+    private static final String TEST_END_DATE_RANGE_MESSAGE = "This order cannot last longer than 12 months";
+    private static final String TEST_UNDER_DATE_RANGE_MESSAGE = "This order must last for at least 1 month";
+
+    private static final int MAXIMUM_MONTHS_ACCEPTED = 12;
+    private static final int MINIMUM_MONTHS_ACCEPTED = 1;
+
+    private final Time time = new FixedTimeConfiguration().stoppedTime();
+    private final LocalDate todayDate = time.now().toLocalDate();
+    private final LocalDate tomorrowDate = time.now().plusDays(1).toLocalDate();
+    private final LocalDate approvalDate = todayDate;
+    private final LocalDateTime approvalDateTime = time.now();
+
+    private final ManageOrderEndDateWithEndOfProceedingsValidator underTest =
+        new ManageOrderEndDateWithEndOfProceedingsValidator(time);
+
+    @Test
+    void accept() {
+        assertThat(underTest.accept()).isEqualTo(MANAGE_ORDER_END_DATE_WITH_END_OF_PROCEEDINGS);
+    }
+
+    // Date
```

Comment:
you could remove comment and if you want to group them, you can use @Nested test class